dashboard/app/models/workshop.rb

```diff
@@ -62,33 +62,46 @@ def phase_info
     ActivityConstants::PHASES[self.phase]
   end
 
-  def program_type_info
-    ActivityConstants::PHASES[self.program_type.to_i]
+  def phase_long_name
+    return '' if phase_info.nil?
+    phase_info[:long_name]
   end
 
-  def self.send_automated_emails
-    [Workshop.workshops_in_2_weeks, Workshop.workshops_in_3_days, Workshop.workshops_ending_today].each do |workshop_list|
-      workshop_list.each do |workshop|
-        teachers = workshop.teachers
-        drop_ins = workshop.unexpected_teachers
-        facilitators = workshop.facilitators
-        district_contacts = workshop.district_contacts
-        [teachers, drop_ins, facilitators, district_contacts].each do |recipient_list|
-          recipient_list.each do |recipient|
-            if EmailValidator::email_address?(recipient.email)
-              if workshop.segments.first.start.to_date == Date.today
```

Comment:
the bug was here.

Compare with the `ending_today` method: that has the correct logic (though it's "workshops that have a last segment that started or will start today" not "ending today").

This logic was "does this workshop that have a first segment that started today (workshop.segments is ordered by start)".